src/Core/View.js

```diff
@@ -141,6 +141,9 @@ function _preprocessLayer(view, layer, provider, parentLayer) {
         };
     }
 
+    layer.getCoords = layer.getCoords || (node => [node.extent]);
```

Comment:
it's necessary that any layer is these functions?

Reply:
I have no clue. It is only a refactor that does not change any functionality (and `getCoordsForLayer` was returning `[node.extent]` as a general case for all layers)

Reply:
with the static functions as seen above you could move this part in this one